_posts/zman7895/2017-05-22-zman7895-turtle-hack.md

```diff
@@ -0,0 +1,18 @@
+---
+layout: post
+author: zman7895
+title: "zman7895's turtle hack"
+---
+Below is my embedded Turtle Hack program.
+<iframe src="https://trinket.io/embed/python/ccad0f5a12" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+I knew from the very beginning I wanted to implement a competitive game for my turtle hack, mainly because I am a competitive person. I decided I was going to do a trivia game between two players. This was all in my mind playing out without being in front of the project. Later in the evening I was watching the NBA Playoffs, and I liked the idea of having the game be a best of 7 like the series are in the NBA. I realized in order to do this, I would need 7 questions, and someone would have to win each round, to make it a best of 7. I didn't know how I was going to do this, but I had my overall idea to sleep on. It was the next day that I sat down in front of my project. I first created the graphic of a black screen with a white finish line. Then I created two turtles to appear as opposing players. I utilized the input method we learned in the exercise to ask players the trivia questions. All of this came pretty easily just off of what we had done in class. Then things got interesting.
+
+
+I needed to learn how to react to right and wrong answers. I went ahead and read about logic and conditionals. I knew I needed to utilize these to have the playing turtles react to right and wrong answers. Figuring this out took forever, and after an hour of trying everything I emailed the professor for help. Only to realize I was simply missing the colon after the IF statement! I was extrememly frustrated with myself but also relieved. Then I spent a lot of time determining what I wanted to happen when players got it right or wrong, and had to figure out how far a turtle should move in order to make it that 4 right answers (or a combination of right answers and wrong answers by the other player) got the player to the finish line. I tried playing my game, and I realized that what I was typing wasn't always giving me the response I wanted. I realized players might answer different variations of the right answer (for example, capitalized answers versus not capitalized answers). My next step was attending to this issue and adding several right answers.
```

Comment:
More good examples of process here, with perhaps the exception of the lost hour.  A little pause somewhere in there, which writing me an email provided, likely would have helped you find the error sooner.

Getting some basic questions implemented was phase one.  That then allowed you to start improving the questions and logic, and identifying the problem of variable user input.